extensions/resource-deployment/src/services/resourceTypeService.ts

```diff
@@ -245,6 +252,21 @@ export class ResourceTypeService implements IResourceTypeService {
 		return undefined;
 	}
 
+	/**
+	 * Get the ok button text based on the selected options
+	 */
+	private getOkButtonText(resourceType: ResourceType, selectedOptions: { option: string, value: string }[]): string | undefined {
+		if (resourceType.okButtonText && selectedOptions.length === 1) {
+			const optionGiven = `${selectedOptions[0].option}=${selectedOptions[0].value}`;
```

Comment:
It looks like we already have logic to go through and parse when clauses in getProvider - please make an issue to combine these as a follow up task.